ui/src/components/app-form/app-form.tsx

```diff
@@ -124,8 +138,8 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
     formState: { errors },
   } = useForm<AppFormInput>({
     defaultValues: {
-      display_name: '',
-      description: '',
+      display_name: formData?.user_options.display_name || '',
+      description: formData?.user_options.description || '',
```

Comment:
These shouldn't be set here, they should be set in the useEffect here: https://github.com/nebari-dev/jhub-apps/blob/adfd708310c970afbe935b49b185eb2fa5174bac/ui/src/components/app-form/app-form.tsx#L342